src/lib/QueryParser.js

```diff
@@ -192,7 +192,7 @@ foam.CLASS({
     {
       name: 'grammar_',
       factory: function() {
-        var cls = foam.lookup(this.of);
+        var cls = this.of$cls;
```

Comment:
I didn't write `Class2` for this PR; let's take up the questions of its design in a meeting.